service.py

```diff
@@ -258,8 +258,10 @@ def playlistIndex(url, playlist):
 ydl.add_default_info_extractors()
 
 with ydl:
-    showInfoNotification("Resolving stream(s) for " + url)
+    progress = xbmcgui.DialogProgressBG()
+    progress.create("Resolving " + url)
     result = ydl.extract_info(url, download=False)
+    progress.close()
```

Comment:
I think progress is not getting closed if we try to play incompatible URL. Progress is getting stuck

Reply:
Can you elaborate? I would have thought that if we get a traceback/crash on the extract line that the python interpreter for the addon is stopped and thus all gui elements get cleared? (I might be wrong if you encountered otherwise)

Reply:
You can replicate the issue after following exactly these steps, step by step
1. Try to open a incompatible url (It will throw error). eg. https://www.mxplayer.in/show/watch-campus-diaries/season-1/episode-2-online-e5a899b3457c2907e0d39f3f4794b35f?utm_source=mx_android_share
2. Now open a working URL (Video will be played): eg. https://youtu.be/qO0ynYGC5iw
3. Now stop the played video.
4. You will see progress of first video on the top right. and it will never disappear.

Recommended solution: Try catch block might help. Not sure.

```
2022-09-18 00:21:51.962 T:24288    INFO <general>: JSONRPC Server: New connection added
2022-09-18 00:22:01.597 T:19416    INFO <general>: initializing python engine.
2022-09-18 00:22:04.265 T:19416   ERROR <general>: ERROR: [Mxplayer] e5a899b3457c2907e0d39f3f4794b35f: Unable to download JSON metadata: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
                                                   
2022-09-18 00:22:04.267 T:19416   ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'lib.yt_dlp.utils.DownloadError'>
                                                   Error Contents: ERROR: [Mxplayer] e5a899b3457c2907e0d39f3f4794b35f: Unable to download JSON metadata: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
                                                   Traceback (most recent call last):
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\extractor\common.py", line 805, in _request_webpage
                                                       return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\YoutubeDL.py", line 3680, in urlopen
                                                       return self._opener.open(req, timeout=self._socket_timeout)
                                                     File "C:\Program Files\WindowsApps\XBMCFoundation.Kodi_19.4.500.0_x64__4n2hpmxwrvr6p\system\python\Lib\urllib\request.py", line 531, in open
                                                       response = meth(req, response)
                                                     File "C:\Program Files\WindowsApps\XBMCFoundation.Kodi_19.4.500.0_x64__4n2hpmxwrvr6p\system\python\Lib\urllib\request.py", line 640, in http_response
                                                       response = self.parent.error(
                                                     File "C:\Program Files\WindowsApps\XBMCFoundation.Kodi_19.4.500.0_x64__4n2hpmxwrvr6p\system\python\Lib\urllib\request.py", line 569, in error
                                                       return self._call_chain(*args)
                                                     File "C:\Program Files\WindowsApps\XBMCFoundation.Kodi_19.4.500.0_x64__4n2hpmxwrvr6p\system\python\Lib\urllib\request.py", line 502, in _call_chain
                                                       result = func(*args)
                                                     File "C:\Program Files\WindowsApps\XBMCFoundation.Kodi_19.4.500.0_x64__4n2hpmxwrvr6p\system\python\Lib\urllib\request.py", line 649, in http_error_default
                                                       raise HTTPError(req.full_url, code, msg, hdrs, fp)
                                                   urllib.error.HTTPError: HTTP Error 400: Bad Request
                                                   
                                                   During handling of the above exception, another exception occurred:
                                                   
                                                   Traceback (most recent call last):
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\extractor\common.py", line 672, in extract
                                                       ie_result = self._real_extract(url)
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\extractor\mxplayer.py", line 135, in _real_extract
                                                       data_json = self._download_json(f'{API_URL}{type}/{video_id}', display_id, headers=headers)['profile']
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\extractor\common.py", line 1032, in download_content
                                                       res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\extractor\common.py", line 996, in download_handle
                                                       res = self._download_webpage_handle(
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\extractor\common.py", line 866, in _download_webpage_handle
                                                       urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\extractor\common.py", line 823, in _request_webpage
                                                       raise ExtractorError(errmsg, cause=err)
                                                   lib.yt_dlp.utils.ExtractorError: Unable to download JSON metadata: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
                                                   
                                                   During handling of the above exception, another exception occurred:
                                                   
                                                   Traceback (most recent call last):
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\YoutubeDL.py", line 1477, in wrapper
                                                       return func(self, *args, **kwargs)
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\YoutubeDL.py", line 1553, in __extract_info
                                                       ie_result = ie.extract(url)
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\extractor\common.py", line 698, in extract
                                                       raise type(e)(e.orig_msg, **kwargs)
                                                   lib.yt_dlp.utils.ExtractorError: [Mxplayer] e5a899b3457c2907e0d39f3f4794b35f: Unable to download JSON metadata: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
                                                   
                                                   During handling of the above exception, another exception occurred:
                                                   
                                                   Traceback (most recent call last):
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\service.py", line 263, in <module>
                                                       result = ydl.extract_info(url, download=False)
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\YoutubeDL.py", line 1466, in extract_info
                                                       return self.__extract_info(url, self.get_info_extractor(key), download, extra_info, process)
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\YoutubeDL.py", line 1495, in wrapper
                                                       self.report_error(str(e), e.format_traceback())
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\YoutubeDL.py", line 996, in report_error
                                                       self.trouble(f'{self._format_err("ERROR:", self.Styles.ERROR)} {message}', *args, **kwargs)
                                                     File "C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\lib\yt_dlp\YoutubeDL.py", line 936, in trouble
                                                       raise DownloadError(message, exc_info)
                                                   lib.yt_dlp.utils.DownloadError: ERROR: [Mxplayer] e5a899b3457c2907e0d39f3f4794b35f: Unable to download JSON metadata: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
                                                   -->End of Python script error report<--
                                                   
2022-09-18 00:22:04.459 T:19416    INFO <general>: Python interpreter stopped
2022-09-18 00:22:15.732 T:11004    INFO <general>: initializing python engine.
2022-09-18 00:22:18.524 T:11004    INFO <general>: plugin.video.sendtokodi: could not find an original manifest or manifest is not supported falling back to best all-in-one stream
2022-09-18 00:22:18.525 T:11004    INFO <general>: plugin.video.sendtokodi: creating list item for url https://rr2---sn-npupj5cax-qxal.googlevideo.com/videoplayback?expire=1663462337&ei=YRcmY_i4AojDrtoPr9WLkAo&ip=103.199.225.236&id=o-AD_Hl7iyaWD1NhlfIm7f-0KyhyE9mmiHPmcqIFhQTYTv&itag=22&source=youtube&requiressl=yes&mh=bF&mm=31%2C29&mn=sn-npupj5cax-qxal%2Csn-qxaelnez&ms=au%2Crdu&mv=m&mvi=2&pl=24&initcwndbps=827500&spc=yR2vp-qRJfrkcMAJ5TOb-8706pG_ovo&vprv=1&svpuc=1&mime=video%2Fmp4&cnr=14&ratebypass=yes&dur=877.667&lmt=1650165215848223&mt=1663440304&fvip=3&fexp=24001373%2C24007246&beids=24277532&c=ANDROID&rbqsm=fr&txp=5532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Ccnr%2Cratebypass%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAJ7BpZ2MCUJWLZrxxhZIp8-vrWisg93AHkUed0diJLBrAiBQIsM0jGwDm3KEeURpL0NHupDCiVLi4pl8a8yGgwXwPw%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRgIhALK36mfG1w_fcPFM1teg2BYVe-6o92E59tNtvXq0uqNuAiEAi45AhyOBUvK1ht7lMRmPpFNv7zfEyhtQg3M1XQFdYvY%3D
2022-09-18 00:22:18.538 T:11004    INFO <general>: CPythonInvoker(8, C:\Users\GG\AppData\Roaming\Kodi\addons\plugin.video.sendtokodi\service.py): script successfully run
2022-09-18 00:22:18.630 T:31184    INFO <general>: VideoPlayer::OpenFile: plugin://plugin.video.sendtokodi/?https://youtu.be/qO0ynYGC5iw
2022-09-18 00:22:18.677 T:11004    INFO <general>: Python interpreter stopped
2022-09-18 00:22:18.688 T:29200    INFO <general>: Creating InputStream
2022-09-18 00:22:18.802 T:29200    INFO <general>: Creating Demuxer
2022-09-18 00:22:18.946 T:29200    INFO <general>: Opening stream: 0 source: 256
2022-09-18 00:22:18.946 T:29200    INFO <general>: Creating video codec with codec id: 27
2022-09-18 00:22:18.946 T:29200    INFO <general>: CDVDVideoCodecFFmpeg::Open() Using codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
2022-09-18 00:22:18.946 T:29200    INFO <general>: Creating video thread
2022-09-18 00:22:18.946 T:26576    INFO <general>: running thread: video_thread
2022-09-18 00:22:18.948 T:29200    INFO <general>: Opening stream: 1 source: 256
2022-09-18 00:22:18.948 T:29200    INFO <general>: Finding audio codec for: 86018
2022-09-18 00:22:18.949 T:29200    INFO <general>: CDVDAudioCodecFFmpeg::Open() Successful opened audio decoder aac
2022-09-18 00:22:18.949 T:29200    INFO <general>: Creating audio thread
2022-09-18 00:22:18.949 T:18944    INFO <general>: running thread: CVideoPlayerAudio::Process()
2022-09-18 00:22:18.949 T:26576    INFO <general>: DXVA::CContext::CreateContext: creating discrete d3d11va device for decoding.
2022-09-18 00:22:18.959 T:18944    INFO <general>: Creating audio stream (codec id: 86018, channels: 2, sample rate: 44100, no pass-through)
2022-09-18 00:22:18.963 T:27304    INFO <general>: CActiveAESink::OpenSink - initialize sink
2022-09-18 00:22:18.986 T:26576    INFO <general>: DXVA::CDecoder::Open: Total video memory available is 10421 MB (dedicated = 495 MB, shared = 9925 MB)
2022-09-18 00:22:19.041 T:31184    INFO <general>: DXVA::CProcessorHD::InitProcessor: supported deinterlace methods: blend:no, bob:yes, adaptive:yes, mocomp:yes.
2022-09-18 00:22:19.042 T:31184 WARNING <general>: CRendererHQ::CheckVideoParameters: chosen scaling method 1 is not supported by renderer
2022-09-18 00:22:19.065 T:31184    INFO <general>: Loading skin file: VideoFullScreen.xml, load type: KEEP_IN_MEMORY
2022-09-18 00:22:19.288 T:26576    INFO <general>: CDVDVideoCodecFFmpeg::CDropControl: calculated diff time: 40000
2022-09-18 00:22:51.054 T:24288    INFO <general>: JSONRPC Server: Disconnection detected
2022-09-18 00:23:01.770 T:24288    INFO <general>: JSONRPC Server: New connection added
2022-09-18 00:23:01.897 T:24288    INFO <general>: JSONRPC Server: Disconnection detected
```